### PR TITLE
configs, board/rtl8721csm: Wlan performance improvement

### PR DESCRIPTION
**Pull request by prashantrar**
This PR contains the following changes:

1. Change wlan thread priority and some driver changes of lib_wlan.a to improve wifi performance.
2. Some changes in LWIP and TCP options in defconfig.

All commits have been labelled with respective changes.

### CODE REVIEW

**Comment by prashantrar** [comment]
@sunghan-chang is there any progress on this PR.

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c [hunk 1]
```diff
@@ -659,7 +659,7 @@ static int _tizenrt_create_task(struct task_struct *ptask, const char *name, u32
 		sparam.sched_priority = 206;
 	}
 	if (strcmp(name, "rtw_recv_tasklet") == 0) {
-		sparam.sched_priority = 205;
+		sparam.sched_priority = 206;
```
Comment (sunghan-chang) [inline os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:662]: Could you let me know why you set it over 200? Normal pri of apps is 100.

Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:662]: @sunghan-chang  This priority setting has been made to achieve the 20Mbps throughput target for RX and TX. these priorities are for our wifi driver tasks.

Comment (jsdosa) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:662]: @prashantrar 
Other tasks may not work properly when Wi-Fi is running.
now, HPWORK priority is 201.

Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:662]: Alright I have tested with the lower priority I.e. in the 100 range and performance is not affected. I will push it soon.

Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c]: I have fixed the priorities now

**Comment by prashantrar** [comment]
@sunghan-chang , @jsdosa The priorities are fixed now, is there anything else that needs to be done for this PR?

**Comment by prashantrar** [comment]
@sunghan-chang is there any update on this PR

**Comment by prashantrar** [comment]
@pillip8282 the permission has been fixed now

**Comment by prashantrar** [comment]
@pillip8282 can this be merged now?

**Pull request merged**